plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoDeleteFile.java

```diff
@@ -0,0 +1,339 @@
+    {
+        return nanValueCounts;
+    }
+
+    @JsonProperty("lowerBounds")
+    @Nullable
+    public Map<Integer, byte[]> lowerBoundsAsByteArray()
+    {
+        return lowerBounds;
+    }
+
+    @Override
+    @Nullable
+    public Map<Integer, ByteBuffer> lowerBounds()
+    {
+        if (lowerBounds == null) {
+            return null;
+        }
+        return lowerBounds.entrySet().stream()
+                .collect(toImmutableMap(Map.Entry<Integer, byte[]>::getKey, entry -> ByteBuffer.wrap(entry.getValue())));
+    }
+
+    @JsonProperty("upperBounds")
+    @Nullable
+    public Map<Integer, byte[]> upperBoundsAsByteArray()
+    {
+        return upperBounds;
+    }
+
+    @Override
+    @Nullable
+    public Map<Integer, ByteBuffer> upperBounds()
+    {
+        if (upperBounds == null) {
+            return null;
+        }
+        return upperBounds.entrySet().stream()
+                .collect(toImmutableMap(Map.Entry<Integer, byte[]>::getKey, entry -> ByteBuffer.wrap(entry.getValue())));
+    }
+
+    @JsonProperty("keyMetadata")
+    @Nullable
+    public byte[] keyMetadataAsByteArray()
+    {
+        return keyMetadata;
+    }
+
+    @Override
+    @Nullable
+    public ByteBuffer keyMetadata()
+    {
+        if (keyMetadata == null) {
+            return null;
+        }
+        return ByteBuffer.wrap(keyMetadata);
+    }
+
+    @Override
+    @JsonProperty("equalityFieldIds")
+    @Nullable
+    public List<Integer> equalityFieldIds()
+    {
+        return equalityFieldIds;
+    }
+
+    @Override
+    @JsonProperty("sortOrderId")
+    @Nullable
+    public Integer sortOrderId()
+    {
+        return sortOrderId;
+    }
+
+    @Override
+    @JsonProperty("splitOffsets")
+    @Nullable
+    public List<Long> splitOffsets()
+    {
+        return splitOffsets;
+    }
+
+    @Override
+    public DeleteFile copy()
+    {
+        return this;
+    }
+
+    @Override
+    public DeleteFile copyWithoutStats()
+    {
+        return new TrinoDeleteFile(
+                pos,
+                specId,
+                fileContent,
+                path,
+                format,
+                recordCount,
+                fileSizeInBytes,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                keyMetadata,
+                equalityFieldIds,
+                sortOrderId,
+                splitOffsets);
+    }
+
+    public long getRetainedSizeInBytes()
+    {
+        ToLongFunction<Integer> intSizeOf = ignored -> Integer.SIZE / 8;
+        ToLongFunction<Long> longSizeOf = ignored -> Long.SIZE / 8;
+        return INSTANCE_SIZE
+                + estimatedSizeOf(path)
+                + estimatedSizeOf(columnSizes, intSizeOf, longSizeOf)
+                + estimatedSizeOf(nullValueCounts, intSizeOf, longSizeOf)
+                + estimatedSizeOf(nanValueCounts, intSizeOf, longSizeOf)
```

Comment:
We should add `estimatedSizeOf(Map<K, V> map, long keySize, long valueSize)` overload where key and value sizes are constant. This would avoid map traversal.

Reply:
You can leverage https://github.com/airlift/slice/pull/146 now.

Reply:
right

Reply:
let's apply https://github.com/trinodb/trino/pull/11642#discussion_r842859668 and release airbase too